botocore/configloader.py

```diff
@@ -142,12 +143,13 @@ def raw_config_parse(config_filename, parse_subsections=True):
         path = os.path.expandvars(path)
         path = os.path.expanduser(path)
         if not os.path.isfile(path):
-            raise botocore.exceptions.ConfigNotFound(path=path)
-        cp = configparser.RawConfigParser()
+            raise botocore.exceptions.ConfigNotFound(path=_unicode_path(path))
+        cp = six.moves.configparser.RawConfigParser()
         try:
-            cp.read(path)
-        except configparser.Error:
-            raise botocore.exceptions.ConfigParseError(path=path)
+            cp.read([path])
```

Comment:
Note that the list was added because of this logic:
https://github.com/python/cpython/blob/master/Lib/configparser.py#L690-L692

If you pass bytes to the ``read``, it will not wrap the filename in a list and just iterate through the individual bytes in the bytes object...